runtime/v2/runc/v1/service.go

```diff
@@ -324,46 +223,41 @@ func (s *service) Create(ctx context.Context, r *taskAPI.CreateTaskRequest) (_ *
 			Terminal: r.Terminal,
 		},
 		Checkpoint: r.Checkpoint,
-		Pid:        uint32(pid),
+		Pid:        uint32(container.Pid()),
 	})
 
 	return &taskAPI.CreateTaskResponse{
-		Pid: uint32(pid),
+		Pid: uint32(container.Pid()),
 	}, nil
-
 }
 
 // Start a process
 func (s *service) Start(ctx context.Context, r *taskAPI.StartRequest) (*taskAPI.StartResponse, error) {
-	p, err := s.getProcess(r.ExecID)
+	container, err := s.getContainer()
 	if err != nil {
 		return nil, err
 	}
 
 	// hold the send lock so that the start events are sent before any exit events in the error case
 	s.eventSendMu.Lock()
-	if err := p.Start(ctx); err != nil {
+	p, err := container.Start(ctx, r)
+	if err != nil {
 		s.eventSendMu.Unlock()
-		return nil, err
+		return nil, errdefs.ToGRPC(err)
 	}
-
-	// case for restore
-	if s.getCgroup() == nil && p.Pid() > 0 {
-		cg, err := cgroups.Load(cgroups.V1, cgroups.PidPath(p.Pid()))
-		if err != nil {
-			logrus.WithError(err).Errorf("loading cgroup for %d", p.Pid())
-		}
-		s.setCgroup(cg)
+	if err := s.ep.Add(container.ID, container.Cgroup()); err != nil {
+		logrus.WithError(err).Error("add cg to OOM monitor")
 	}
-	if r.ExecID != "" {
+	switch r.ExecID {
+	case "":
 		s.send(&eventstypes.TaskExecStarted{
-			ContainerID: s.id,
+			ContainerID: container.ID,
 			ExecID:      r.ExecID,
```

Comment:
Shouldn't this be in `default` as `ExecID` will be empty?

Reply:
Same question in start function in runtime/v2/runc/v2/service.go

Reply:
oops, I got this backwards